### PR TITLE
TF v2.12 and misc. bug fixes

### PR DESCRIPTION
**Pull request by tuncK**
@anuprulez and @bgruening,

I attempted to update this image (and the tool ). Mostly due to old lib versions, I was unable to use it for a project due to errors here and there (lib missing, permission denied, file not found...) .

I also tried to address https://github.com/usegalaxy-eu/issues/issues/450 by forcing GPU version of TF to be used if an NVIDIA GPU is detected during the startup.

### CODE REVIEW

**Comment by bgruening** [comment]
Nice! I think we should add CI testing here and automatic pushes. You can get inspiration from the normal jupyter notebook I think, if you have time to work on it.

**Comment by tuncK** [comment]
I will work on it.

Question, likely beyond this PR: why do we have 2 separate jupyter notebooks?

Would it not be possible to combine ML and normal jupyter notebooks into 1 xml only and switch between them, if necessary, by choosing from a drop menu?

**Comment by bgruening** [comment]
> Would it not be possible to combine ML and normal jupyter notebooks into 1 xml only and switch between them, if necessary, by choosing from a drop menu?

That is a great idea. But I don't know how to do that yet. Requirements (that is the container) are fixed and can not be changed at runtime yet.
(History: the previous iteration called IEs had this feature)

**Comment by bgruening** [comment]
Oh sorry, I had hoped we had something more recent in the other repo: https://github.com/bgruening/docker-jupyter-notebook/blob/master/.travis.yml

But I was wrong. I guess you need to start from scratch :(

**Comment by tuncK** [comment]
I added GH workflow for merges and PRs, also set the logins as secrets. Though as if it does not take into effect immediately?

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v3
+          
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+        
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with:
```
Comment (bgruening) [inline .github/workflows/pr.yml:23]: we need to push to quaq.io.

I think we can use `registry: quay.io` according to https://medium.com/@erdal.genc09/github-action-to-automate-build-push-for-dockerfiles-repository-to-quay-io-docker-registry-9ebbf6007e2

What I don't understand is ... I only have a `OAUTH_TOKEN` not a username. But I guess I can try a few option and see if that works.

File: .github/workflows/merge.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  build_and_push:
+    needs: initialise
+    runs-on: ubuntu-latest
+    
+    steps:
+      - name: Build docker image and push online
+        uses: docker/build-push-action@v4
+        with:
+          push: true
+          tags: ktunc/docker-ml-jupyterlab:latest
```
Comment (bgruening) [inline .github/workflows/merge.yml]: this might need to become:

```suggestion
          tags: quay.io/galaxy/docker-ml-jupyterlab:latest
```

Comment (bgruening) [reply .github/workflows/merge.yml]: And do we want to version it? So not use latest?

Comment (tuncK) [reply .github/workflows/merge.yml]: Currently, it is on Anup's account with manual versioning (latest=0.2).

Putting to Galaxy's institutional account makes sense. There is no hard-coded version anywhere in this repo, can we maybe version like this:
`docker tag quay.io/galaxy/docker-ml-jupyterlab:$(git rev-parse --short HEAD)"`

Comment (bgruening) [reply .github/workflows/merge.yml]: I added a QUAY_OAUTH_TOKEN to the secrets.

We could push on release-tags. So only if we create a tag we actually push.

Comment (tuncK) [reply .github/workflows/merge.yml]: I changed it to quay, on release, as you suggested. Lets see if it works.

**Comment by bgruening** [comment]
Please feel free to create a new dummy PR, changing a whitespace, see if it turns green, merge and tag a release. I'm going to bed here :-)

**Comment by bgruening** [comment]
Thanks a lot @tuncK this looks very cool!

**Pull request merged**